sasdata/data_util/manipulations.py

```diff
@@ -362,19 +362,45 @@ def _avg(self, data2D, maj):
         qy_data = data2D.qy_data[np.isfinite(data2D.data)]
         mask_data = data2D.mask[np.isfinite(data2D.data)]
 
+        # Was getting negative bin widths if whole box was below Qx = 0 for maj == 'x' or Qy = 0 for maj == 'y'.
+        # Should this raise error or correct bin width?
+        if self.bin_width < 0:
+            # raise RuntimeError("_Slab._avg: bin width can't be negative")
+            self.bin_width = abs(self.bin_width)
+
         # Build array of Q intervals
         if maj == 'x':
             if self.fold:
-                x_min = 0
+                # Set x_max based on which is further from Qx = 0
+                x_max = max(abs(self.x_min),abs(self.x_max))
+                # Set x_min based on which is closer to Qx = 0, but will have different limits depending on whether
+                # x_min and x_max are on the same size of Qx = 0
+                if self.x_min*self.x_max >= 0: # If on same side
+                    x_min = min(abs(self.x_min),abs(self.x_max))
+                else:
+                    x_min = 0
             else:
+                x_max = self.x_max
                 x_min = self.x_min
-            nbins = int(math.ceil((self.x_max - x_min) / self.bin_width))
+            y_max = self.y_max
+            y_min = self.y_min
+            nbins = int(math.ceil((x_max - x_min) / self.bin_width))
         elif maj == 'y':
             if self.fold:
-                y_min = 0
+                # Set y_max based on which is further from Qy = 0
+                y_max = max(abs(self.y_min), abs(self.y_max))
+                # Set y_min based on which is closer to Qy = 0, but will have different limits depending on whether
+                # y_min and y_max are on the same size of Qy = 0
+                if self.y_min * self.y_max >= 0:  # If on same side
+                    y_min = min(abs(self.y_min), abs(self.y_max))
+                else:
+                    y_min = 0
             else:
+                y_max = self.y_max
                 y_min = self.y_min
-            nbins = int(math.ceil((self.y_max - y_min) / self.bin_width))
+            x_max = self.x_max
+            x_min = self.x_min
+            nbins = int(math.ceil((y_max - y_min) / self.bin_width))
```

Comment:
Could all of the above be simplified by taking the absolute value of `y_max - y_min` here and the absolute value of `x_max - x_min` on line 387? The only time you would then need to check max x/y is for times where 0 is in your range and `self.fold`.

```suggestion
            if self.fold and self.y_min * self.y_max <= 0:   # y == 0 in range
                    # Set y_max based on which is further from Qy = 0
                    y_max = max(abs(self.y_min), abs(self.y_max))
                    y_min = 0
            else:
                y_max = self.y_max
                y_min = self.y_min
            nbins = int(math.ceil(abs(y_max - y_min) / self.bin_width))
```

Reply:
Not without changing following calculations. This would cause problems in the case that `self.fold` and the box is entirely within either the negative x when `maj == 'x'` or the negative y when `maj == 'y'`. In this situation, proposed line 407 and 408 would cause the max and min to be negative which is undesirable when folding.